onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -22,33 +22,35 @@ struct OpenVINOProviderFactory : IExecutionProviderFactory {
   std::string device_type_;
   bool enable_vpu_fast_compile_;
   std::string device_id_;
+  size_t num_of_threads_;
 };
 
 std::unique_ptr<IExecutionProvider> OpenVINOProviderFactory::CreateProvider() {
-  OpenVINOExecutionProviderInfo info(device_type_, enable_vpu_fast_compile_, device_id_);
+  OpenVINOExecutionProviderInfo info(device_type_, enable_vpu_fast_compile_, device_id_, num_of_threads_);
   return std::make_unique<OpenVINOExecutionProvider>(info);
 }
 
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_OpenVINO(
-    const char* device_type, bool enable_vpu_fast_compile, const char* device_id) {
-  return std::make_shared<onnxruntime::OpenVINOProviderFactory>(device_type, enable_vpu_fast_compile, device_id);
+    const char* device_type, bool enable_vpu_fast_compile, const char* device_id, size_t num_of_threads) {
+  return std::make_shared<onnxruntime::OpenVINOProviderFactory>(device_type, enable_vpu_fast_compile, device_id, num_of_threads);
 }
 
 }  // namespace onnxruntime
 
 ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProvider_OpenVINO,
                     _In_ OrtSessionOptions* options, _In_ const char* device_type) {
   options->provider_factories.push_back(
-      onnxruntime::CreateExecutionProviderFactory_OpenVINO(device_type, false, ""));
+      onnxruntime::CreateExecutionProviderFactory_OpenVINO(device_type, false, "", 8));
   return nullptr;
 }
 
-ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProviderEx_OpenVINOEP,
+ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProviderEx_OpenVINO,
```

Comment:
>ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProviderEx_OpenVINO [](start = 0, length = 71)

this is a good catch for the fix. should we add some test to cover this in future?

Reply:
@HectorSVC I will create a separate ticket later to cover up this and follow up with our team. Will make this part of our CI Pipeline as well, so that this would be covered.